src/lib/pka_queue.c

```diff
@@ -155,6 +155,33 @@ static uint16_t pka_concat6_wlen(uint32_t word_len,
     return total_wlen;
 }
 
+// Adjusts the value present in the array pointed by buf_ptr.
+// Adjustment is done by removing zero padding at the most significant bytes.
+// Once this is done, align the new length around 8-byte boundary.
+// Note: Here it's assumed that the value is in Little Endian format.
+static void pka_adjust_rslt(pka_operand_t *result)
+{
+    // First find the most significant byte based upon the len, and
+    // then move backwards over all zero bytes.
+    uint8_t  *byte_ptr;
+    uint32_t  result2_len;
+
+    result2_len = result->actual_len;
+    byte_ptr    = result->buf_ptr;
+
+    // Result from pka hardware is always in Little Endian.
+    byte_ptr = &byte_ptr[result2_len - 1];

```

Comment:
This is true. But is it always true? what if the hardware change? Can we consider an if statement that handles both BE and LE. I think the byte order is stored somewhere, thus we can easily retrieve the information and apply the right logic.

Reply:
I'll add the BE logic, Thanks.